src/icrc1_auction_api_mock.mo

```diff
@@ -330,26 +331,34 @@ actor class Icrc1AuctionAPI(adminPrincipal_ : ?Principal) = self {
     return null;
   };
 
-  public shared query ({ caller }) func queryTokenBids(ledger : Principal) : async [(Auction.OrderId, Order)] = async switch (getAssetId(ledger)) {
-    case (?aid) {
-      U.unwrapUninit(auction).getOrders(caller, #bid, ?aid)
-      |> Array.tabulate<(Auction.OrderId, Order)>(_.size(), func(i) = mapOrder(_ [i]));
+  public shared query ({ caller }) func queryTokenBids(ledger : Principal) : async ([(Auction.OrderId, Order)], Nat) {
+    let orders = switch (getAssetId(ledger)) {
```

Comment:
Isn't this switch better done with a let-else statement? Then we can return right away.

Reply:
done